src/components/CheckboxInput/CheckboxInput.js

```diff
@@ -87,6 +89,11 @@ export const CheckboxInput = ({
       : `${styles.CheckboxInput}`
   }
 
+  /**
+   * If consumer's used the brute force `checked` prop, we prioritize that
+   */
+  const resolvedIsChecked = typeof checked !== 'undefined' ? checked : isChecked
```

Comment:
So, my thought is that this is a purely additive change to the EDS checkbox component so there should be no risk of regression in terms of preserving previous functionality to existing usages across our Ethos apps. Put differently, if `checked` prop is not explicitly used by the consumer, everything else should be same as it was where `isChecked` dictates if the checkbox is checked or not.